pandas/core/indexes/datetimelike.py

```diff
@@ -670,8 +670,9 @@ def __add__(self, other):
             elif isinstance(self, TimedeltaIndex) and isinstance(other, Index):
                 if hasattr(other, '_add_delta'):
                     return other._add_delta(self)
-                raise TypeError("cannot add TimedeltaIndex and {typ}"
-                                .format(typ=type(other)))
+                raise TypeError("cannot add {cls} and {typ}"
+                                .format(cls=type(self).__name__,
+                                        typ=type(other).__name__))
```

Comment:
I am not sure this is an improvement in robustness, as *if* series would use the same code, this 'cls' would become either "Series" or "TimedeltaArray", and "Series" is not that informative, and "TimedeltaArray" should not bubble up to user when using a Series (that would be an internal implementation detail).

Reply:
The above is *always* a TimedeltaIndex given the elif statement above